salt/step-ca/salt-client.sls

```diff
@@ -0,0 +1,115 @@
+#
+# step-ca-formula
+#
+# Copyright (C) 2025   darix
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+#
+# Copyright (C) 2022 SUSE LLC
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+{%-   set salt_root_cert = "/etc/salt/step/config/root.crt" %}
+
+{%- set salt_client = false %}
+{%- if 'step_salt_client' in pillar %}
+  {%- set salt_client = pillar.step_salt_client %}
+{%- endif %}
+
+{%- if salt_client %}
```

Comment:
the logic here is kinda backwards.e.g. the step-cli  is needed in both cases

Reply:
I see in original ca.sls we install step-ca , step-cli is installed via client.sls ,
I strart to wonder if make more sense to update/edit client.sls , for example make 
```
step_dir = "/etc/step"
```
to be pillar var with default to /etc/step , also we need to add one additional cfg for ca password .. 

IMO this is will the best approach .. and look like will requires less changes from what I did so far